types/Ints_hasgo.go

```diff
@@ -218,6 +218,22 @@ func (s Ints) Init() (out Ints) {
 	return slicecopy[:len(s)-1]
 }
 
+// =============== inits.go =================
+
+// Inits returns all inits of a sequence, in order of small to large, as if it were called recursively.
+// Can be generated for any type.
+func (s Ints) Inits() (out [][]int64) {
+	out = append(out, make(Ints, 0))
+	for i := range s {
+		new := make(Ints, i+1)
+		for n := 0; n <= i; n++ {
+			new[n] = s[n]
```

Comment:
I know Go allows to use `new` as a variable, but it's also a keyword. To avoid confusion I'd use a different variable name here. (Somebody might assume `new[n]` is some kind of syntax)